### PR TITLE
Allow either SEX_OFFENSE or SEX_OFFENDER in charge_category field

### LINKED ISSUE #609: Allow filters to receive SEX_OFFENDER OR SEX_OFFENSE values in charge_category
**What needs to be done? Why does it need to be done?**
The BE is transitioning from value of SEX_OFFENDER to SEX_OFFENSE in the charge_category field. We need to ensure that both of these values can pass through the filters and return a match, such that when the BE value changes from SEX_OFFENDER to SEX_OFFENSE the numbers on the FE remain correct.

**Additional context**
Add any other context about the task here.

### PR DESCRIPTION
**Pull request by jovergaag**
## Description of the change

The backend is going to be transitioning from using the term SEX_OFFENDER to SEX_OFFENSE in the next week or two. This change is preparing for that transition, to ensure that when the data changes the dashboard will continue to display the correct data, even when that filter is applied.

This PR will be followed up with a second PR, to remove the SEX_OFFENDER value, and use SEX_OFFENSE instead. The issue for that work is #610 

## Type of change

- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Configuration change (adjusts configuration to achieve some end related to functionality, development, performance, or security)

## Related issues

> Closes #609 

### Development

These boxes should be checked by the submitter prior to merging:

- [x] Manual testing against realistic data has been performed locally

### Code review

These boxes should be checked by reviewers prior to merging:

- [x] This pull request has a descriptive title and information useful to a reviewer
- [x] This pull request has been moved out of a Draft state, has no "Work In Progress" label, and has assigned reviewers
- [x] Potential security implications or infrastructural changes have been considered, if relevant

### CODE REVIEW

**Comment by jovergaag** [comment]
attn: @jjdressel11 @caroletouma

File: src/components/charts/new_revocations/helpers.js [hunk 1]
```diff
@@ -68,8 +68,10 @@ export const matchesTopLevelFilters = ({
     (dimensionKey === undefined || dimensionKey === "charge_category") &&
     filters[CHARGE_CATEGORY] &&
     !skippedFilters.includes(CHARGE_CATEGORY) &&
-    !(treatCategoryAllAsAbsent && isAllItem(filters[CHARGE_CATEGORY])) &&
-    !nullSafeComparison(item.charge_category, filters[CHARGE_CATEGORY])
+    !(Array.isArray(filters[CHARGE_CATEGORY])
+      ? filters[CHARGE_CATEGORY].includes(item.charge_category)
+      : (treatCategoryAllAsAbsent && isAllItem(filters[CHARGE_CATEGORY])) ||
+        nullSafeComparison(item.charge_category, filters[CHARGE_CATEGORY]))
```
Comment (macfarlandian) [inline src/components/charts/new_revocations/helpers.js:74]: this logic is pretty hard to follow inline ... what would you think about factoring it out into a function?

File: src/components/charts/propTypes.js [hunk 2]
```diff
@@ -9,7 +9,7 @@ export const metricTypePropType = PropTypes.oneOf([
 export const filtersPropTypes = PropTypes.shape({
   metricPeriodMonths: PropTypes.oneOfType([PropTypes.string, PropTypes.number])
     .isRequired,
-  chargeCategory: PropTypes.string.isRequired,
+  chargeCategory: PropTypes.oneOfType([PropTypes.string, PropTypes.array]),
```
Comment (macfarlandian) [inline src/components/charts/propTypes.js:12]: same PropType comment as above applies here as well. also, why no longer required?

File: src/views/tenants/constants/filterOptions.js [hunk 3]
```diff
@@ -83,11 +83,14 @@ const MOFilterOptions = {
     filterEnabled: false,
     componentEnabled: false,
   },
+  // temporarily we will be accepting either SEX_OFFENSE or SEX_OFFENDER
+  // in the charge_category field. Once the BE transition to SEX_OFFENSE
+  // has been made, we will revert this to the single value
```
Comment (macfarlandian) [inline src/views/tenants/constants/filterOptions.js:88]: this comment appears a bunch of times, it might be helpful to tag it with a `TODO(#xxx)` to help find them all later?

Comment (jovergaag) [reply src/views/tenants/constants/filterOptions.js:88]: Good idea.

**Review by macfarlandian — APPROVED** [review]
🎉
